test/SchedulerTest.cpp

```diff
@@ -1,71 +1,80 @@
 #include "pipert/Scheduler.h"
 #include "gtest/gtest.h"
 
-#include <iostream>
-#include <utility>
-
 class Human {
 public:
   Human(const std::string& name, int age) : name_(name), age_(age) {}
 
   std::string GetName() const { return name_; }
   int GetAge() const { return age_; }
 
-private: 
+private:
   std::string name_;
   int age_;
 };
 
-class Printer {
-public:
-  void Print(pipert::PacketToProcess<Human> packet) {
-    std::cout << packet.data().GetName() << " - " <<  packet.data().GetAge() << std::endl;
-  }
-};
+using SchedulerDeathTest = pipert::Scheduler;
+
+template <class T>
+class SchedulerTest : public testing::Test {};
+
+using MyTypes = ::testing::Types<char, int, Human>;
+TYPED_TEST_SUITE(SchedulerTest, MyTypes, );
+
+void SchedulerInitTestWithParameter(int workers) {
+  pipert::Scheduler sch(workers);
+  exit (0);
+}
+
+void DoNothing() {
+  return;
+}
 
-/*
-TEST(SchedulerTest, SchedulerState) {
-    pipert::Scheduler sch1(0);
-    pipert::Scheduler sch2(2);
+TEST(SchedulerDeathTest, InitializationWithParameter) {
+#ifdef DEBUG
+  EXPECT_DEATH(SchedulerInitTestWithParameter(-5), ".*");
```

Comment:
Can I implement this logic in Scheduler? Then I could drop these test cases and test the new functionality.

Reply:
Yes. I think you can put this avoidance behavior. I think that the expectation is to
 - assert in debug to the user telling that something stupid was done
 - but still do something non-fatal, meaningful in release
So this shows that your test should only test the release behavior (that it is surviving).
I think that such fatal crashes are allowed in scenarios where it would slow down the performance of the library. This case is special because this functionality is used in the setup phase before the system start its real-time processes.

Reply:
Created a patch for this in my latest changes.